src/Shared/EnumUtilities.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Build.Shared
+{
+    public static class EnumUtilities
+    {
+        private static readonly Dictionary<Enum, string> _enumStringCache = [];
+
+        public static string GetEnumString(Enum value)
```

Comment:
I have very split feelings about it.
Waring - following is very opinionated:
 * If we do not want to optimise - then we should not and just use the ToString
 * If we DO want to optimise (very likely the case here) - then we should create a simple efficient targetted switch method or compile-time initialized array - but just for that single enum type.
 * Throwing everything in a singe global singleton might perform very great initially, but if it starts to be used on multiple places - then it can start to suffer. So why creating a catch-all singleton if we actually do not want to use it that much?

Reply:
I'd say don't add this class at all. Especially not as new public API.